src/core/network.js

```diff
@@ -432,6 +592,104 @@ var NetworkManager = (function NetworkManagerClosure() {
+      PDFNetworkStreamFullRequestReader_parseContentDisposition(string) {
+        if (!string || typeof string !== 'string') {
+          throw new InvalidHeaderException('argument string is required');
+        }
+
+        var contentDisposition = {};
+        var match = DISPOSITION_TYPE_REGEXP.exec(string);
+        if (!match) {
+          throw new InvalidHeaderException('invalid type format');
+        }
+
+        // normalize type
+        var index = match[0].length;
+        contentDisposition.type = match[1].toLowerCase();
+
+        var key;
+        var names = [];
+        var params = {};
+        var value;
+
+        // calculate index to start at
+        index = PARAM_REGEXP.lastIndex = match[0].substr(-1) === ';' ?
+          index - 1 : index;
+
+        // match parameters
+        while ((match = PARAM_REGEXP.exec(string))) {
+          if (match.index !== index) {
+            throw new InvalidHeaderException('invalid parameter format');
+          }
+
+          index += match[0].length;
+          key = match[1].toLowerCase();
+          value = match[2];
+
+          if (names.indexOf(key) !== -1) {
+            throw new InvalidHeaderException('invalid duplicate parameter');
+          }
+
+          names.push(key);
+
+          if (key.indexOf('*') + 1 === key.length) {
+            // decode extended value
+            key = key.slice(0, -1);
+            value = decodefield(value);
+
+            // overwrite existing value
+            params[key] = value;
+            continue;
+          }
+
+          if (typeof params[key] === 'string') {
+            continue;
+          }
+
+          if (value[0] === '"') {
+            // remove quotes and escapes
+            value = value
+              .substr(1, value.length - 2)
+              .replace(QESC_REGEXP, '$1');
+          }
+
+          params[key] = value;
+        }
+
+        if (index !== -1 && index !== string.length) {
+          throw new InvalidHeaderException('invalid parameter format');
+        }
+
+        contentDisposition.parameters = params;
+        return contentDisposition;
+    },
+
+    _parseHeaders: function PDFNetworkStreamFullRequestReader_parseHeaders() {
+      try {
+        var headers = Object.create(null);
+        var networkManager = this._manager;
+        var fullRequestXhrId = this._fullRequestId;
+        var fullRequestXhr = networkManager.getRequestXhr(fullRequestXhrId);
+        var contentDispositionString =
+          fullRequestXhr.getResponseHeader('Content-Disposition');
+
+        if (contentDispositionString && contentDispositionString.length > 0) {
+          headers.contentDisposition =
+            this._parseContentDisposition(contentDispositionString);
+        }
+
+        return headers;
+      } catch(err) {
+        if (err instanceof InvalidHeaderException) {
+          // Ignore invalid headers
+          return headers;
+        } else {
+          throw err;
+        }
+      }
+    },
+
```

Comment:
This chunk of code is unnecessarily complex. We are only interested in the file name, so don't bother with parsing the rest. Get rid of the exceptions and return as soon as you have found a filename.

Once you've determined the file name, save it in this PDFNetworkStreamFullReader instance as `.filename`.